frontend/server/src/Controllers/Tag.php

```diff
@@ -65,4 +65,37 @@ public static function getPublicTags() {
             'problemTag'
         );
     }
+
+    /**
+     * Return most frequent public tags of a certain level
+     *
+     * @return list<array{alias: string}>
+     */
+    public static function getFrequentTagsByLevel(
+        string $problemLevel
+    ): array {
+        return \OmegaUp\Cache::getFromCacheOrSet(
+            \OmegaUp\Cache::TAGS_LIST,
+            "level-{$problemLevel}",
+            fn () => \OmegaUp\DAO\Tags::getFrequentTagsByLevel(
+                $problemLevel
+            ),
+            APC_USER_CACHE_SESSION_TIMEOUT
+        );
+    }
+
+    /**
+     * Return most frequent public tags of a certain level
+     *
+     * @return array{frequent_tags: list<array{alias: string}>}
+     *
+     * @omegaup-request-param string $problemLevel
+     */
+    public static function apiFrequentTags(\OmegaUp\Request $r): array {
+        $param = $r->ensureString('problemLevel');
```

Comment:
```suggestion
        $param = $r->ensureString(
            'problemLevel',
            fn (string $problemAlias) => \OmegaUp\Validators::alias($problemAlias));
```

así si nos intentan pasar algo que no es un alias, lo rechazamos.